File: Domain/Post.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set;}
        public string Title {get; set; }
        public string Main2 { get; set; }
        public string Main { get; set; }
        public string Description { get; set; }
         public string Category { get; set; }
         public string Link { get; set; }
         public string Image { get; set; }
        public DateTime Date { get; set; }
        public virtual ICollection<UserPost> UserPosts {get; set;}
        public virtual ICollection<PostComment> PostComments{get; set;}
           
    }
}

File: client-app/src/App/Models/post.ts
export interface IPostsEnvelope {
    posts: IPost[];
    postCount: number;
  }

export interface IPost {
    id: string;
    title: string;
    description: string;
    image: string;
    main2: string;
    main: string;
    category: string;
    date: Date;
    isHost: boolean;
    isLiked: boolean;
    liked: ILikedPost[];
    postComments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
  }

export interface IPostFormValues extends Partial<IPost> {
    time?: Date;
}
export class PostFormValues implements IPostFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = "";
    description: string = "";
    date?: Date = undefined;
    time?: Date = undefined;
    main?: string = '';
    main2?: string = '';
    image?: string = '';
    link?: string = '';


    constructor(init?: IPostFormValues) {
        if (init && init.date){
            init.time = init.date
        }
        Object.assign(this, init);
    } 
}

export interface ILikedPost {
    username: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean;
}